releasenotes-builder/src/test/java/com/github/checkstyle/MainTest.java

```diff
@@ -0,0 +1,217 @@
+                "-githubAuthToken", "TOKEN",
+                "-generateAll",
+                "-publishTwit",
+                "-twitterConsumerKey", "KEY",
+                "-twitterConsumerSecret", "SECRET",
+                "-twitterAccessToken", "TOKEN",
+                "-twitterAccessTokenSecret", "SECRET",
+                "-validateVersion"
+        );
+    }
+
+    @Test
+    public void testIssueCommitWithIssueNotFound() throws IOException {
+        TEST_COMMITS.add(RevCommitUtil.create("Issue #1: Hello World"));
+
+        assertMainReturnCode(-2,
+                "-localRepoPath", temporaryFolder.newFolder().getAbsolutePath(),
+                "-remoteRepoPath", "checkstyle/checkstyle",
+                "-startRef", "12345678",
+                "-releaseNumber", "10.0.1",
+                "-outputLocation", temporaryFolder.newFolder().getAbsolutePath(),
+                "-githubAuthToken", "TOKEN",
+                "-generateAll",
+                "-publishTwit",
+                "-twitterConsumerKey", "KEY",
+                "-twitterConsumerSecret", "SECRET",
+                "-twitterAccessToken", "TOKEN",
+                "-twitterAccessTokenSecret", "SECRET",
+                "-validateVersion"
+        );
+    }
+
+    @Test
+    public void testIssueCommit() throws Exception {
+        TEST_COMMITS.add(RevCommitUtil.create("Issue #1: Hello World"));
+        TEST_ISSUES.add(GhIssueUtil.create(1, GHIssueState.OPEN, "Hello World",
+                Constants.MISCELLANEOUS_LABEL));
+
+        assertMainReturnCode(0,
+                "-localRepoPath", temporaryFolder.newFolder().getAbsolutePath(),
+                "-remoteRepoPath", "checkstyle/checkstyle",
+                "-startRef", "12345678",
+                "-releaseNumber", "10.0.1",
+                "-outputLocation", temporaryFolder.newFolder().getAbsolutePath(),
+                "-githubAuthToken", "TOKEN",
+                "-generateAll",
+                "-publishTwit",
+                "-twitterConsumerKey", "KEY",
+                "-twitterConsumerSecret", "SECRET",
+                "-twitterAccessToken", "TOKEN",
+                "-twitterAccessTokenSecret", "SECRET",
+                "-validateVersion"
+        );
+    }
+
+    /**
+     * Helper method to run {@link Main#main(String...)} and verify the exit code.
+     * Uses {@link Mockito#mockStatic(Class)} to mock method {@link Runtime#exit(int)}
+     * to avoid VM termination.
+     *
+     * @param expectedExitCode the expected exit code to verify
+     * @param arguments the command line arguments
+     * @noinspection CallToSystemExit, ResultOfMethodCallIgnored
+     * @noinspectionreason CallToSystemExit - test helper method requires workaround to
+     *      verify exit code
+     * @noinspectionreason ResultOfMethodCallIgnored - temporary suppression until #11589
+     */
+    private static void assertMainReturnCode(int expectedExitCode, String... arguments) {

```

Comment:
runMainAndAssertReturnCode

Reply:
Done, also noting this same method came from main repo.